.github/workflows/build-and-push-docker-image.yml

```
@@ -0,0 +1,33 @@
+---
+
+name: 🏗️
+
+on:   # yamllint disable-line rule:truthy
+  pull_request:
+  push:
+    branches: ["release/*", "unstable/*"]
+    tags: ["*"]
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+    - uses: actions/checkout@v4
+    - name: Build Docker image
+      run: |
+        IMAGE="ghcr.io/$GITHUB_REPOSITORY:${GITHUB_REF_NAME/'/'/'-'}"
+        echo "IMAGE=$IMAGE" >>"$GITHUB_ENV"
+        docker build . \
+          --build-arg BUILDKIT_INLINE_CACHE=1 \
+          --cache-from $IMAGE \
+          --tag $IMAGE
+    - name: Log in to GHCR
+      if: github.event_name != 'pull_request'
+      run: |
```

I usually prefer this syntax so that the whole thing is parsed as a single line:
```suggestion
      run: >-
```

---

Generally I don't use `run: >-` because it will fold the contents of the block into a single line. This can alter the behavior of the script within the block. On this line though, it makes no difference, so sure.